src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGenerator.cs

```diff
@@ -1,30 +1,54 @@
 ﻿using System;
 using System.Diagnostics;
 using System.IO;
+using System.Net.Mime;
 using System.Text;
-using Uno.SourceGeneration;
+using Microsoft.CodeAnalysis;
+using Uno.Roslyn;
 using Uno.UI.SourceGenerators.Telemetry;
 
+#if NETFRAMEWORK
+using Uno.SourceGeneration;
+#endif
+
 namespace Uno.UI.SourceGenerators.XamlGenerator
 {
+#if NETFRAMEWORK
 	[GenerateAfter("Uno.UI.SourceGenerators.DependencyObject." + nameof(DependencyObject.DependencyPropertyGenerator))]
-	public class XamlCodeGenerator : SourceGenerator
+#endif
+	[Generator]
+	public class XamlCodeGenerator : ISourceGenerator
 	{
-		public override void Execute(SourceGeneratorContext context)
+		public void Initialize(GeneratorInitializationContext context)
 		{
-			var gen = new XamlCodeGeneration(
-				context.Compilation,
-				context.GetProjectInstance(),
-				context.Project
-			);
+		}
+
+		public void Execute(GeneratorExecutionContext context)
+		{
+			DependenciesInitializer.Init(context);
+
+			// No initialization required for this one
+			// if(!Process.GetCurrentProcess().ProcessName.Equals("devenv", StringComparison.OrdinalIgnoreCase))
+			{
+				// Debugger.Launch();
+			}
+
+			var gen = new XamlCodeGeneration(context);
 
 			if (PlatformHelper.IsValidPlatform(context))
 			{
 				var genereratedTrees = gen.Generate();
 
 				foreach (var tree in genereratedTrees)
 				{
-					context.AddCompilationUnit(tree.Key, tree.Value);
+					context.AddSource(tree.Key, tree.Value);
+
+					var intermediatePath = context.GetMSBuildPropertyValue("IntermediateOutputPath");
```

Comment:
Codacy found an issue: [Remove the unused local variable 'intermediatePath'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6270956)